packages/bsky/src/api/app/bsky/unspecced/getPostThreadV2.ts

```diff
@@ -0,0 +1,154 @@
+  SkeletonFnInput,
+  createPipeline,
+  noRules,
+} from '../../../../pipeline'
+import { postUriToThreadgateUri } from '../../../../util/uris'
+import { Views } from '../../../../views'
+import { ATPROTO_REPO_REV, resHeaders } from '../../../util'
+
+export default function (server: Server, ctx: AppContext) {
+  const getPostThread = createPipeline(
+    skeleton,
+    hydration,
+    noRules, // handled in presentation: 3p block-violating replies are turned to #blockedPost, viewer blocks turned to #notFoundPost.
+    presentation,
+  )
+  server.app.bsky.unspecced.getPostThreadV2({
+    auth: ctx.authVerifier.optionalStandardOrRole,
+    handler: async ({ params, auth, req, res }) => {
+      const { viewer, includeTakedowns, include3pBlocks } =
+        ctx.authVerifier.parseCreds(auth)
+      const labelers = ctx.reqLabelers(req)
+      const hydrateCtx = await ctx.hydrator.createContext({
+        labelers,
+        viewer,
+        includeTakedowns,
+        include3pBlocks,
+      })
+
+      let result: OutputSchema
+      try {
+        result = await getPostThread({ ...params, hydrateCtx }, ctx)
+      } catch (err) {
+        const repoRev = await ctx.hydrator.actor.getRepoRevSafe(viewer)
+        if (repoRev) {
+          res.setHeader(ATPROTO_REPO_REV, repoRev)
+        }
+        throw err
+      }
+
+      const repoRev = await ctx.hydrator.actor.getRepoRevSafe(viewer)
+
+      return {
+        encoding: 'application/json',
+        body: result,
+        headers: resHeaders({
+          repoRev,
+          labelers: hydrateCtx.labelers,
+        }),
+      }
+    },
+  })
+}
+
+const skeleton = async (inputs: SkeletonFnInput<Context, Params>) => {
+  const { ctx, params } = inputs
+  const anchor = await ctx.hydrator.resolveUri(params.anchor)
+  try {
+    const res = await ctx.dataplane.getThread({
+      postUri: anchor,
+      above: params.above,
+      below: calculateBelow(ctx, anchor, params),
+    })
+    return {
+      anchor,
+      uris: res.uris,
+    }
+  } catch (err) {
+    if (isDataplaneError(err, Code.NotFound)) {
+      return {
+        anchor,
+        uris: [],
+      }
+    } else {
+      throw err
+    }
+  }
+}
+
+const hydration = async (
+  inputs: HydrationFnInput<Context, Params, Skeleton>,
+) => {
+  const { ctx, params, skeleton } = inputs
+  return ctx.hydrator.hydrateThreadPosts(
+    skeleton.uris.map((uri) => ({ uri })),
+    params.hydrateCtx,
+  )
+}
+
+const presentation = (
+  inputs: PresentationFnInput<Context, Params, Skeleton>,
+) => {
+  const { ctx, params, skeleton, hydration } = inputs
+  const thread = ctx.views.threadV2(skeleton, hydration, {
+    above: params.above,
+    below: calculateBelow(ctx, skeleton.anchor, params),
+    branchingFactor: params.branchingFactor,
+    prioritizeFollowedUsers: params.prioritizeFollowedUsers,
+    sorting: params.sorting,
+    viewer: params.hydrateCtx.viewer,
+  })
+
+  const anchor = thread.find((i) => i.depth === 0)
```

Comment:
No big deal here, but it feels like there should be a more direct way to check whether the thread was "found" other than scanning the thread items for the anchor post.

I also wonder if it's a bit more direct to check based on the anchor's uri rather than knowing that it's the unique item with `depth: 0`.